hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/FieldProbe.java

```diff
@@ -79,28 +70,30 @@ String getProbeName() {
     }
 
     static <S> FieldProbe createFieldProbe(Field field, Probe probe, SourceMetadata sourceMetadata) {
-        int type = getType(field.getType());
-        if (type == -1) {
+        ProbeType type = getType(field.getType());
+        if (type == null) {
             throw new IllegalArgumentException(format("@Probe field '%s' is of an unhandled type", field));
         }
 
-        if (isDouble(type)) {
+        if (type.getMapsTo() == double.class) {
             return new DoubleFieldProbe<S>(field, probe, type, sourceMetadata);
-        } else {
+        } else if (type.getMapsTo() == long.class) {
             return new LongFieldProbe<S>(field, probe, type, sourceMetadata);
+        } else {
+            throw new IllegalArgumentException(type.toString());
         }
     }
 
     static class LongFieldProbe<S> extends FieldProbe implements LongProbeFunction<S> {
 
-        LongFieldProbe(Field field, Probe probe, int type, SourceMetadata sourceMetadata) {
+        LongFieldProbe(Field field, Probe probe, ProbeType type, SourceMetadata sourceMetadata) {
             super(field, probe, type, sourceMetadata);
         }
 
         @Override
         public long get(S source) throws Exception {
             switch (type) {
-                case TYPE_PRIMITIVE_LONG:
+                case TYPE_LONG_PRIMITIVE:
```

Comment:
Once the type system is enriched with the other primitive types, you can use a VarHandle and use the same technique to prevent litter as with the MethodProbe.

Reply:
> Once the type system is enriched with the other primitive types, you can use a VarHandle and use the same technique to prevent litter as with the MethodProbe.

I thought it already did for some reason.

[Raised separately](https://github.com/hazelcast/hazelcast/issues/25528), although a very easy change.

Reply:
What is needed here is something like this:

```
  static class LongMethodProbe<S> extends MethodProbe implements LongProbeFunction<S> {

        LongMethodProbe(Method method, Probe probe, int type, SourceMetadata sourceMetadata) {
            super(method, probe, type, sourceMetadata);
        }

        @Override
        public long get(S source) throws Exception {
            switch (type) {
                case TYPE_PRIMITIVE_LONG:
                    return (long) METHOD_HANDLE.invokeExact(source);
                case TYPE_PRIMITIVE_INT:
                    return (int) METHOD_HANDLE.invokeExact(source);
                case TYPE_PRIMITIVE_BYTE:
                    return (byte) METHOD_HANDLE.invokeExact(source);
                case TYPE_PRIMITIVE_CHAR:
                    return (char) METHOD_HANDLE.invokeExact(source);
                ....    
                    ```

Reply:
I think the type enrichment can be left for a future PR